pkg/controller/direct/registry/registry.go

```diff
@@ -46,6 +49,14 @@ func GetModel(gk schema.GroupKind) (directbase.Model, error) {
 	return registration.model, nil
 }
 
+func GetRolloutGate(gk schema.GroupKind) (predicate.RolloutGate, error) {
+	registration := singleton.registrations[gk]
+	if registration == nil || registration.model == nil {
+		return nil, fmt.Errorf("no model registered for %s", gk)
```

Comment:
I'm not sure that this should be an error (?)

Reply:
Yeah, fair enough. Changed it to not return any error, just nil if there is no rollout gate set.